src/middlewared/middlewared/plugins/jail.py

```diff
@@ -460,16 +456,17 @@ def fstab(self, jail, options):
         _pass = options.get('pass')
         index = options.get('index')
 
-        if verrors:
-            raise verrors
-
         if action == 'replace' and index is None:
-            raise ValueError(
-                "Index must not be None when replacing fstab entry"
+            verrors.add(
+                'index',
+                'Index must not be None when replacing fstab entry'
             )
 
+        if verrors:
+            raise verrors
+
         _list = iocage.fstab(action, source, destination, fstype, fsoptions,
-                             dump, _pass, index=index, add_path=True)
+                             dump, _pass, index=index)
```

Comment:
Why removing add_path this time? What does it do?

Reply:
So what add_path does is that if set to true it automatically adds the iocroot to the destination. Now we are allowing users two options. To either specify full path or from the jails root. In both cases this solution would work

Reply:
`            if add_path:
                destination = f"{self.iocroot}/jails/{uuid}/root{destination}"`

Reply:
This snippet is from iocage